vendor/bat-native-ads/src/bat/ads/internal/ad_server/catalog/catalog_creative_set_info.cc

```diff
@@ -35,4 +35,21 @@ bool CatalogCreativeSetInfo::operator!=(
   return !(*this == rhs);
 }
 
+bool CatalogCreativeSetInfo::DoesSupportOS() const {
+  if (oses.empty()) {
+    // Creative set supports all operating systems
+    return true;
+  }
+
+  const std::string platform_name = PlatformHelper::GetInstance()->GetName();
+
+  for (const auto& os : oses) {
```

Comment:
nit: should we use std::find_if here?

Reply:
agreed and we have another open issue to fix these, thanks